src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -389,7 +390,12 @@ public override async Task<bool> AreSynchronized(Uri firstResourceUri, Uri secon
             {
                 if (await source.ExistsAsync())
                 {
-                    return !string.IsNullOrEmpty(source.Properties.ContentMD5) && source.Properties.ContentMD5 == destination.Properties.ContentMD5;
+                    var sourceBlobMetadata = source.Metadata;
```

Comment:
Is this code path tested?

Reply:
Yeah, the "ExistsAsync()" will fetch the properties and metadata so we can get the metadata later as it is. Verify it locally.
The [document](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-container-properties-metadata) is here.